src/Bicep.Core/PrettyPrint/DocumentBuildVisitor.cs

```diff
@@ -148,6 +148,26 @@ public ILinkedDocument BuildDocument(SyntaxBase syntax)
                 this.Visit(syntax.Value);
             });
 
+        public override void VisitParameterAssignmentSyntax(ParameterAssignmentSyntax syntax) =>
```

Comment:
>VisitParameterAssignmentSyntax [](http://example.com/codeflow?start=29&length=30)

Good catch. We should probably build a custom analyzer that's aware of our visitor pattern to catch stuff like this.

Reply:
Another possibility is to verify that the pre- and post- pretty-print files are syntactically (excluding whitespace) equivalent.

Reply:
For this visitor, definitely. I was thinking of catching similar issues in all/most visitors, including any future ones.

Reply:
Hard to really verify programmatically because the majority of visitors will only override a few methods, by design. I was wondering about switching this one to use the interface rather than abstract base class, as it really feels like it requires implementing almost all base methods.